controllers/vspherecluster_controller.go

```diff
@@ -213,6 +200,12 @@ func (r clusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr er
 func (r clusterReconciler) reconcileDelete(ctx *context.ClusterContext) (reconcile.Result, error) {
 	ctx.Logger.Info("Reconciling VSphereCluster delete")
 
+	// If the cluster is deleted, that's mean that the workload cluster is being deleted and so the CCM/CSI instances
+	if !ctx.Cluster.DeletionTimestamp.IsZero() {
```

Comment:
Nit: no longer need the IF condition

Reply:
What if the vSphereCluster gets deleted but the Cluster not? is that a use case we want to support?
In that case CCM/CSI technically could continue to live in the workload cluster, even if there is nothing reconciling them...

Reply:
we should remove the if, as the ccm and csi needs the control plane endpoint (which is removed when the `VSphereCluster` is)